### PR TITLE
Feat/refactoring and test examples

### PR DESCRIPTION
**Pull request by Ruslan719**

### CODE REVIEW

File: NotadogApi.Tests/Hubs/GameHubTests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using NUnit.Framework;
+
+namespace NotadogApi.Tests.Hubs
+{
+	class GameHubTests
+	{
+		#region Fields
+
+		private GameHub _gameHub;
+
```
Comment (klikkn) [inline NotadogApi.Tests/Hubs/GameHubTests.cs]: do we need empty rows between class vars?

File: NotadogApi.Tests/UnitTest1.cs [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+
+		// Runs before every test.
+		[SetUp]
+	    public void SetUp()
+		{
+			result = true;
+		}
+
+	    [Test]
+        public void Mehtod_Conditions_ExpectedAction()
```
Comment (klikkn) [inline NotadogApi.Tests/UnitTest1.cs:18]: we can just remove it maybe

File: NotadogApi/Controllers/UsersController.cs [hunk 3]
```diff
@@ -37,12 +37,12 @@ public UsersController(IUserService userService, IJwtTokenGenerator jwtTokenGene
         [HttpPost("login")]
         public async Task<IActionResult> Login(UserLoginCredentials credentials)
         {
-            var user = await _userService.GetOneByEmailAsync(credentials.Email);
+            var user = await _userService.GetOneByEmailAsync(credentials.Email.Trim());
```
Comment (klikkn) [inline NotadogApi/Controllers/UsersController.cs]: `var trimmedEmail = credentials.Email.Trim()` ?

File: NotadogApi/Controllers/UsersController.cs [hunk 4]
```diff
@@ -37,12 +37,12 @@ public UsersController(IUserService userService, IJwtTokenGenerator jwtTokenGene
...
         public async Task<IActionResult> Login(UserLoginCredentials credentials)
         {
-            var user = await _userService.GetOneByEmailAsync(credentials.Email);
+            var user = await _userService.GetOneByEmailAsync(credentials.Email.Trim());
 
             if (user == null)
                 return NotFound();
 
-            if (user.Password != credentials.Password)
+            if (user.Password != credentials.Password.Trim())
```
Comment (klikkn) [inline NotadogApi/Controllers/UsersController.cs]: var trimmedPassword ?)

File: NotadogApi/Controllers/UsersController.cs (hunk 3 above)
Comment (klikkn) [reply NotadogApi/Controllers/UsersController.cs]: I think we don't need trim on BE.
we probably can add exception if `credentials .email.trim != credentials .email`

File: NotadogApi/Hubs/GameHubMethod.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿namespace NotadogApi.Hubs
+{
+	public enum GameHubMethod
```
Comment (klikkn) [inline NotadogApi/Hubs/GameHubMethod.cs:3]: cool!

**Review by klikkn — CHANGES_REQUESTED** [review]

File: NotadogApi/Controllers/UsersController.cs (hunk 3 above)
Comment (klikkn) [reply NotadogApi/Controllers/UsersController.cs]: I think later we will add full validation for email, name and password. Email validation will include trim checking

**Review by enfacee — DISMISSED** [review]

File: NotadogApi.Tests/UnitTest1.cs (hunk 2 above)
Comment (Ruslan719) [reply NotadogApi.Tests/UnitTest1.cs:18]: We should remove it later. Now it works as template and explanation.

File: NotadogApi.Tests/Hubs/GameHubTests.cs (hunk 1 above)
Comment (Ruslan719) [reply NotadogApi.Tests/Hubs/GameHubTests.cs]: No, we don't. But why shouldn't i add them? We don't have conventions.

Comment (klikkn) [reply NotadogApi.Tests/Hubs/GameHubTests.cs]: I think we need to discuss it

File: NotadogApi/Controllers/UsersController.cs (hunk 3 above)
Comment (Ruslan719) [reply NotadogApi/Controllers/UsersController.cs]: Not a good idea to throw exception. if user inserts space accidentally, better to correct him. Also between BE and FE I prefer first one to implement.

File: NotadogApi/Hubs/GameHubMethod.cs (hunk 5 above)
Comment (Ruslan719) [reply NotadogApi/Hubs/GameHubMethod.cs:3]: Every time you use the same string literal think about enums or constants.

**Review by klikkn — APPROVED** [review]

**Pull request merged**